src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -325,6 +329,72 @@ EnsureCitusTableCanBeCreated(Oid relationOid)
 }
 
 
+/*
+ * undistribute_table gets a distributed table name and 
+ * udistributes the table.
+ */
+Datum
+undistribute_table(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+
+	Oid relationId = PG_GETARG_OID(0);
+	const char *relationName = get_rel_name(relationId);
```

Comment:
```suggestion
	const char *relationName = get_rel_name(relationId);
```

We should move this to after `relation_open`. This because another backend (session) may rename the table before we acquire the lock and this would yield using a wrong table name.